warp10/src/main/java/io/warp10/script/functions/TOLONGBYTES.java

```diff
@@ -49,13 +51,49 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         } else {
           stack.push(Longs.toByteArray(((Long) o).longValue()));
         }
+      } else if (o instanceof List && nbBytes > 0 && nbBytes <= 8) {
+        try {
+          byte[] b = new byte[nbBytes * ((List) o).size()];
+          ByteBuffer bytes = ByteBuffer.wrap(b);
+          bytes.order(ByteOrder.BIG_ENDIAN);
+          if (8 == nbBytes) {
+            for (Object el: (List) o) {
+              bytes.putLong((Long) el);
+            }
+          } else if (4 == nbBytes) {
+            for (Object el: (List) o) {
+              bytes.putInt(((Long) el).intValue());
+            }
+          } else if (2 == nbBytes) {
+            for (Object el: (List) o) {
+              bytes.putShort(((Long) el).shortValue());
+            }
+          } else if (1 == nbBytes) {
+            for (Object el: (List) o) {
+              bytes.put(((Long) el).byteValue());
+            }
+          } else {
+            // weird length are not optimized
+            long v;
+            for (Object el: (List) o) {
+              v = ((Long) el).longValue() << (8 - nbBytes) * 8;
+              for (int i = 0; i < nbBytes; i++) {
+                bytes.put((byte) (v >> 56));
```

Comment:
Do an unsigned shift instead

Reply:
Why ? I don't care about Long MSB to be set to 0 or 1. unsigned shift vs standard shift is just a different assembly instruction, I doubt there is a clock cycle gain.

Reply:
you're right indeed